src/winconpty/winconpty.cpp

```diff
@@ -48,6 +48,37 @@ static wchar_t* _ConsoleHostPath()
         modulePath.replace_filename(L"OpenConsole.exe");
         if (!std::filesystem::exists(modulePath))
         {
+            std::wstring_view architectureInfix{};
+            USHORT unusedImageFileMachine{}, nativeMachine{};
+            if (IsWow64Process2(GetCurrentProcess(), &unusedImageFileMachine, &nativeMachine))
+            {
+                // Despite being a machine type, the values IsWow64Process2 returns are *image* types
+                switch (nativeMachine)
+                {
+                case IMAGE_FILE_MACHINE_AMD64:
+                    architectureInfix = L"x64";
+                    break;
+                case IMAGE_FILE_MACHINE_ARM64:
+                    architectureInfix = L"arm64";
+                    break;
+                case IMAGE_FILE_MACHINE_I386:
+                    architectureInfix = L"x86";
+                    break;
+                default:
+                    break;
+                }
+            }
+            if (architectureInfix.empty())
+            {
+                // WHAT?
+                return _InboxConsoleHostPath();
+            }
+            modulePath.replace_filename(architectureInfix);
+            modulePath.append(L"OpenConsole.exe");

```

Comment:
Shouldn't this be `L"\\OpenConsole.exe"`?
Also: Should we wrap this code in a feature check? It seems very specific to our nuget version...

Reply:
filesystem::path::append does the right thing here.

In addition, this logic (1) starts with and (2) falls back to, the original code. If you were using conpty.dll before (which you should not have been, as it was not being distributed!) it will still look for OpenConsole.exe in the same folder *first*, and it will still fall back to conhost *last*.

There is no feature flag that will allow us to build this code differently based on whether it is inside the nuget package or not, and I would rather not introduce that complexity.

Reply:
Now that I am at a full non-phone keyboard, here's why I think it would add complexity.

Right now, our feature flags support *branding* and *branches*. A Branding is pretty much which store package we're building, or "windows inbox" for conhost. That's another way to "brand" our codebase. Branches are branches obviously :D

Brandings are *mutually exclusive*. You can't activate two brandings at once. They're supposed to represent a single unified product flow, from Dev->Preview->Stable.

Introducing feature flags for _the conpty package_ within that existing infrastructure requires us to add a new branding. "This is the terminal repo, built for ConPTY_Package". However, we lose a bunch of flexibility here:

1. We would not be able to produce a build with the same code, that is, OpenConsole.exe and winconpty.lib, as a Terminal release or a WPF control release
2. We would not be able to have "Preview" conpty features (unless we added a second new branding, ConPTY_Package_Preview.)
3. We would need to compile the code three *more* times per build to activate that branding on only the components that it matters for. Right now, our build matrix is "[Win10, Win11] * [x86, x64, arm64] * [Preview, Stable]". That's 12 builds on a good day. I don't want to add more.

Reply:
And it wouldn't get us much. It would remove a middle-tier fallback to look for a right-architecture conpty host.